controller/watch/dynamic_resource_client.go

```diff
@@ -0,0 +1,22 @@
+package watch
+
+import (
+	"k8s.io/client-go/1.5/dynamic"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/pkg/watch"
+)
+
+type dynamicResourceClient interface {
+	Watch(*v1.ListOptions) (watch.Interface, error)
+}
+
+type realDynamicResourceClient dynamic.ResourceClient
```

Comment:
just curious - is it common trick to re-type types? can you please help me understand how it benefits the test, or  testability?